### ISSUE TITLE
Chatbot markdown displays multi-line code blocks incorrectly

### ISSUE THREAD

**Issue by Obs01ete** [gradio-app/gradio#3776]
### Describe the bug

Chatbot markdown does not display newlines inside a code block.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

The following string is displayed okay except that the code inside the code quotes (``` blah ```) is displayed without newlines:

mystring = "Solution: To preprocess the historical stock data, we can perform the following steps:\n\n1. Remove any unnecessary columns that do not contribute to the prediction, such as the stock symbol or date.\n2. Check for and handle any missing or null values in the data.\n3. Normalize the data to ensure that all features are on the same scale. This can be done using techniques such as Min-Max scaling or Z-score normalization.\n4. Split the data into training and testing sets. The training set will be used to train the machine learning model, while the testing set will be used to evaluate its performance.\n\nHere is an example code snippet to preprocess the data using Pandas:\n\n```\nimport pandas as pd\nfrom sklearn.preprocessing import MinMaxScaler\nfrom sklearn.model_selection import train_test_split\n\n# Read in the historical stock data\ndata = pd.read_csv('historical_stock_data.csv')\n\n# Remove unnecessary columns\ndata = data.drop(['symbol', 'date'], axis=1)\n\n# Handle missing values\ndata = data.fillna(method='ffill')\n\n# Normalize the data\nscaler = MinMaxScaler()\ndata = scaler.fit_transform(data)\n\n# Split the data into training and testing sets\nX_train, X_test, y_train, y_test = train_test_split(data[:, :-1], data[:, -1], test_size=0.2, random_state=42)\n```\n\nNext request."

### Screenshot

![image](https://user-images.githubusercontent.com/32495260/230442425-6a97c1bb-6f15-4a76-ba95-297975f98691.png)

### Logs

```shell
No error message
```

### System Info

```shell
Windows 10, Chrome

python -c "import gradio; print(gradio.__version__)"
3.23.0
```

### Severity

serious, but I can work around it

**Comment by abidlabs** [comment]
Hi @Obs01ete thanks for creating this issue. I believe this issue falls under #3531. I'll close this issue, and feel free to chime into that issue so that we can deal with this comprehensively.
